cdc/split.cc

```diff
@@ -24,6 +24,203 @@
 
 #include "split.hh"
 
+struct atomic_column_update {
+    column_id id;
+    atomic_cell cell;
+};
+
+struct nonatomic_column_deletion {
```

Comment:
Just curious why we need specifically nonatomic_column_deletion, how does it differ from an atomic one? Are you including deletion of atomic columns as just an atomic_update? (if so, maybe a comment clarifying this can be helpful).

Reply:
The main reason I separated this structure from `nonatomic_column_update` (instead of keeping the tombstone inside `nonatomic_column_update` together with the cells) is for the type system to protect me from accidentally putting cells of a nonatomic column into a `clustered_row_insert`. `clustered_row_insert` doesn't have any `nonatomic_column_update`s, as you can see in the struct definition below.

The explanation why do this is fairly complicated. When in CQL we do
```insert into tab (pk,ck,coll) values (0,0,{...})```
where `coll` is a nonatomic collection column and `{...}` is a collection of this type, this creates a tombstone and a bunch of cells. 
However, if we want to create just a tombstone, there's a way with `insert`:
```insert into tab (pk,ck,coll) values (0,0,null)```
but there is no way to create just the cells with an `insert`. To do that, you have to use `update`:
```update tab set coll = coll + {...} where pk = 0 and ck = 0```

Due to this difference of power of expression between `insert` and `update` I decided that for consistency, I will always split the deletion from the cell addition. Thus, this `insert`:
```insert into tab (pk,ck,coll) values (0,0,{...})```
will be split into two changes in CDC: 
```
insert into tab (pk,ck,coll) values (0,0,null)
update tab set coll = coll + {...}
```
there's another reason to do the split above: the tombstone has a different timestamp than the cells (the difference is `-1`).

I agree this deserves a comment somewhere.

Reply:
And now I see I did write a comment inside `clustered_row_insert` definition. But I'll add a reference here too.